mla/test_statistics.py

```diff
@@ -291,6 +289,17 @@ def threeml_ps_test_statistic(params: np.ndarray,
     """
     temp_params = rf.unstructured_to_structured(
         params, dtype=prepro.params.dtype, copy=True)
+
+    sob_energy = prepro.event_model.get_energy_sob(
+        prepro.events[prepro['drop_index']])
+
+    sob = prepro.sob_spatial * _sob_time(params, prepro) * sob_energy
+
     sob = prepro.sob_spatial * _sob_time(
-        temp_params, prepro) * prepro.sob_energy
+        temp_params, prepro) * sob_energy
+    if 'ns' in temp_params.dtype.fields:
```

Comment:
I think this works, but use `names` instead of `fields` to be sure and to match the rest of the code